### PR TITLE
fix(pipettes): update pressure sensor interrupt for stm32g4

### PR DESCRIPTION
**Pull request by caila-marashaj**
Since the transition to the STM32G4 for the EVT pipettes, we need to make a couple of changes to our interrupt service routine:

- change the EXTI input line from `EXTI9_5_IRQn` to `EXTI3_IRQn`, to correspond with the change in the pressure sensor's data ready pin from `GPIO_PIN_9` to `GPIO_PIN_3`
- change the HAL function that gets implemented in our firmware from `HAL_GPIO_EXTI_Falling_Callback` to `HAL_GPIO_EXTI_Callback`, since the source code is slightly different between the STM32L5 and the STM32G4

This pr just allows the interrupt code in `pipettes/firmware/main.cpp` to be called and executed as expected.

### CODE REVIEW

File: pipettes/firmware/hardware_config.c [hunk 1]
```diff
@@ -42,8 +42,9 @@ IRQn_Type get_interrupt_line(const PipetteType pipette_type) {
         case SINGLE_CHANNEL:
         case EIGHT_CHANNEL:
         default:
-            // External interrupt lines 15:10
-            return EXTI9_5_IRQn;
+            // External interrupt line 3
+            return EXTI3_IRQn;
+//            return EXTI9_5_IRQn;
```
Comment (Laura-Danielle) [inline pipettes/firmware/hardware_config.c:47]: Let's get rid of the commented out line here.

File: pipettes/firmware/main.cpp [hunk 2]
```diff
@@ -105,12 +105,22 @@ static auto pins_for_sensor =
 auto sensor_hardware =
     sensors::hardware::SensorHardware(pins_for_sensor.primary);
 
+/*
+ * Callback for STM32L5xx, we may want to remove this later
+ *
 extern "C" void HAL_GPIO_EXTI_Falling_Callback(uint16_t GPIO_Pin) {
```
Comment (Laura-Danielle) [inline pipettes/firmware/main.cpp:111]: We can remove this. We already have a `main.cpp` file for the L5 board.

File: pipettes/firmware/main.cpp [hunk 3]
```diff
@@ -171,8 +181,9 @@ auto main() -> int {
     adc_init();
     initialize_enc(PIPETTE_TYPE);
 
+    //    iwdg_refresh();
     delay_start(500);
-
+    //    iwdg_refresh();
```
Comment (Laura-Danielle) [inline pipettes/firmware/main.cpp:186]: Remove?

**Review by Laura-Danielle — COMMENTED** [review]
Overall looks good, but let's clean this up real quick.

**Review by Laura-Danielle — APPROVED** [review]
ty!

**Review by sfoster1 — APPROVED** [review]
Great! Thanks for updating the pr description

**Pull request merged**